File: server/src/main/java/com/fluffy/comment/domain/ExamCommentRepository.java
package com.fluffy.comment.domain;

import com.fluffy.global.exception.NotFoundException;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ExamCommentRepository extends JpaRepository<ExamComment, Long> {

    default ExamComment findByIdOrThrow(Long id) {
        return findById(id)
                .orElseThrow(() -> new NotFoundException("존재하지 않는 댓글입니다. 댓글 식별자: %d".formatted(id)));
    }
}

File: server/src/main/kotlin/com/fluffy/comment/domain/ExamCommentRepository.kt
package com.fluffy.comment.domain

import com.fluffy.global.exception.NotFoundException
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.findByIdOrNull

interface ExamCommentRepository : JpaRepository<ExamComment, Long>, ExamCommentRepositoryCustom

fun ExamCommentRepository.findByIdOrThrow(id: Long): ExamComment {
    return findByIdOrNull(id)
        ?: throw NotFoundException("존재하지 않는 댓글입니다. 댓글 식별자: $id")
}